CvGameCoreDLL_Expansion2/CvPlot.cpp

```diff
@@ -15487,53 +15538,35 @@ int CvPlot::GetStrategicValue(PlayerTypes ePlayer) const
 	{
 		CvPlot* pLoopNearbyPlot = iterateRingPlots(this, i);
 
-		if (pLoopNearbyPlot != NULL && pLoopNearbyPlot->isRevealed(eTeam))
-		{
-			//impassable is uninteresting
-			if (pLoopNearbyPlot->isImpassable(eTeam))
-				continue;
-
-			if (pLoopNearbyPlot->isWater())
-			{
-				if (!pLoopNearbyPlot->isLake())
-					iNearbyCoast++;
-
-				continue;
-			}
+		if (pLoopNearbyPlot == NULL)
+			continue;
 
-			if (!pLoopNearbyPlot->isOwned())
-			{
-				iNearbyUnownedLand++;
-			}
-			else if (pLoopNearbyPlot->getOwner() != ePlayer && GET_PLAYER(pLoopNearbyPlot->getOwner()).isMajorCiv())
-			{
-				int iStrengthMultiplier = 1;
-				int iApproachMultiplier = 1;
+		if (!pLoopNearbyPlot->isRevealed(eTeam))
+			continue;
 
-				CivApproachTypes eApproach = pDiplomacyAI->GetCivStrategicApproach(pLoopNearbyPlot->getOwner());
+		//impassable is uninteresting
+		if (!pLoopNearbyPlot->isValidMovePlot(pLoopNearbyPlot->getOwner()))
+			continue;
 
-				if (eApproach == CIV_APPROACH_NEUTRAL)
-					iApproachMultiplier = 2;
-				else if (eApproach > CIV_APPROACH_WAR && eApproach < CIV_APPROACH_NEUTRAL)
-					iApproachMultiplier = 4;
-				else if (eApproach == CIV_APPROACH_WAR)
-					iApproachMultiplier = 8;
+		if (pLoopNearbyPlot->isWater())
+		{
+			if (!pLoopNearbyPlot->isLake())
+				iNearbyCoast++;
 
-				StrengthTypes eStrengthRelativeToUs = pDiplomacyAI->GetMilitaryStrengthComparedToUs(pLoopNearbyPlot->getOwner());
-				if (eStrengthRelativeToUs == STRENGTH_STRONG)
-					iStrengthMultiplier = 2;
-				else if (eStrengthRelativeToUs == STRENGTH_POWERFUL)
-					iStrengthMultiplier = 4;
-				else if (eStrengthRelativeToUs == STRENGTH_IMMENSE)
-					iStrengthMultiplier = 8;
+			continue;
```

Comment:
So we always continue in case nearbyPlot is water

Reply:
yeah, water tiles are not interesting

Reply:
currently simplifying the logic a lot